### PR TITLE
Removes instance database record when provision saga fails

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

**Comment by bnaecker** [comment]
This should address #805 when merged

File: nexus/src/sagas.rs [hunk 1]
```diff
@@ -514,6 +517,53 @@ async fn sic_create_instance_record(
...
+    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
+
+    // We currently only support deleting an instance if it is stopped or
+    // failed, so update the state accordingly to allow deletion.
+    let runtime_state =
+        sagactx.lookup::<InstanceRuntimeState>("initial_runtime")?;
+    let runtime_state = db::model::InstanceRuntimeState {
+        state: db::model::InstanceState::new(InstanceState::Failed),
+        // Must update the generation, or the database query will fail.
+        gen: db::model::Generation::from(runtime_state.gen.next()),
```
Comment (smklein) [inline nexus/src/sagas.rs:536]: This basically has the implicit assumption that "the only other thing which can move us away from the `Creating` state is either the saga completing or this undo action", right?

**Review by smklein — APPROVED** [review]

File: nexus/src/sagas.rs [hunk 2]
```diff
@@ -514,6 +517,53 @@ async fn sic_create_instance_record(
     Ok(instance.runtime().clone().into())
 }
 
+async fn sic_delete_instance_record(
+    sagactx: ActionContext<SagaInstanceCreate>,
+) -> Result<(), anyhow::Error> {
+    println!("A");
```
[the remark this answers is not in the document]
Comment (bnaecker) [reply nexus/src/sagas.rs:523]: Just an easter egg :) Thanks, will remove.

File: nexus/src/sagas.rs (hunk 1 above)
Comment (bnaecker) [reply nexus/src/sagas.rs:536]: That's correct. I tried to make that clear in the following block comment on the `assert!(updated, ...)` line. Though I like your phrasing: "This generation can only be moved forward by the saga's successful completion, or backward here."

**Pull request merged**